litmus-portal/frontend/src/components/WorkflowStepper/index.tsx

```diff
@@ -136,12 +141,82 @@ const CustomStepper = () => {
     (state: RootState) => state.selectTemplate.isDisable
   );
   const userRole = useSelector((state: RootState) => state.userData.userRole);
-
   const tabs = useActions(TabActions);
   const workflow = useActions(WorkflowActions);
   const [invalidYaml, setinValidYaml] = React.useState(false);
   const steps = getSteps();
 
+  function EditYaml() {
+    const oldParsedYaml = YAML.parse(yaml);
+    let NewLink: string = ' ';
+    let NewYaml: string = ' ';
+    if (
+      oldParsedYaml.kind === 'Workflow' &&
+      scheduleType.scheduleOnce !== 'now'
+    ) {
+      NewLink = workflowsList[parseInt(id, 10)].chaosWkfCRDLink_Recur as string;
+      fetch(NewLink)
+        .then((data) => {
+          data.text().then((yamlText) => {
+            const oldParsedYaml = YAML.parse(yaml);
+            const newParsedYaml = YAML.parse(yamlText);
+            delete newParsedYaml.spec.workflowSpec;
+            newParsedYaml.spec.schedule = cronSyntax;
+            delete newParsedYaml.metadata.generateName;
+            newParsedYaml.metadata.name = workflowData.name;
+            newParsedYaml.spec.workflowSpec = oldParsedYaml.spec;
+            NewYaml = YAML.stringify(newParsedYaml);
+            workflow.setWorkflowDetails({
+              link: NewLink,
+              yaml: NewYaml,
+            });
+          });
+        })
+        .catch((err) => {
+          console.error(`Unable to fetch the yaml text${err}`);
+        });
+    }
+    if (
+      oldParsedYaml.kind === 'CronWorkflow' &&
+      scheduleType.scheduleOnce === 'now'
+    ) {
+      NewLink = workflowsList[parseInt(id, 10)].chaosWkfCRDLink as string;
+      fetch(NewLink)
+        .then((data) => {
+          data.text().then((yamlText) => {
+            const oldParsedYaml = YAML.parse(yaml);
+            const newParsedYaml = YAML.parse(yamlText);
+            delete newParsedYaml.spec;
+            delete newParsedYaml.metadata.generateName;
+            newParsedYaml.metadata.name = workflowData.name;
+            newParsedYaml.spec = oldParsedYaml.spec.workflowSpec;
+            NewYaml = YAML.stringify(newParsedYaml);
+            workflow.setWorkflowDetails({
+              link: NewLink,
+              yaml: NewYaml,
+            });
+          });
+        })
+        .catch((err) => {
+          console.error(`Unable to fetch the yaml text${err}`);
+        });
+    }
+    if (
+      oldParsedYaml.kind === 'CronWorkflow' &&
+      scheduleType.scheduleOnce !== 'now'
+    ) {
+      const newParsedYaml = YAML.parse(yaml);
+      newParsedYaml.spec.schedule = cronSyntax;
+      delete newParsedYaml.metadata.generateName;
```

Comment:
I had a talk with @ksatchit regarding this @arkajyotiMukherjee the workflows in chaos charts have more use cases apart from portal, using a generate name will allow the user to run then without the need of deleting old ones with the same name, which will be the problem when we use a static `name` instead of `generateName` so it was finally decided by @rajdas98 to parse and replace after the fetch in frontend.